hack/platform/partials/main.go

```diff
@@ -896,16 +907,63 @@ spec:
 	})
 
 	util.DefaultRequire = false
+	schema := &jsonschema.Schema{}
+	schemaBytes, err := os.ReadFile(jsonSchemaPath)
+	if err != nil {
+		panic(err)
+	}
+	err = json.Unmarshal(schemaBytes, schema)
+	if err != nil {
+		panic(err)
+	}
+	externalProperty, ok := schema.Properties.Get("external")
 
-	paths := []string{
-		"external/platform/apiKey",
-		"external/platform/autoSleep",
-		"external/platform/autoDelete",
-		"external/platform",
-		"external",
+	if !ok {
+		panic("external property not found in " + jsonSchemaPath)
 	}
+	walkTree(externalProperty, schema, "external", "")
 
+	// fmt.Println("properties:")
+	// for childNode := schema.Properties.Oldest(); childNode != nil; childNode = childNode.Next() {
+	// 	fmt.Printf("%v : %v\n", childNode.Key, childNode.Value)
+	// }
+	// fmt.Println(paths)
 	for _, p := range paths {
-		util.GenerateFromPath(util.GenerateSchema(&extconfig.Config{}), util.BasePath+"/config", p, nil)
+		p := strings.TrimPrefix(p, "/")
+		util.GenerateFromPath(schema, util.BasePath+"/config", p, nil)
+	}
+}
+
+var paths []string
+
+func walkTree(node, parent *jsonschema.Schema, name, parentName string) bool {
+	if node == nil || getChildren(node, parent) == nil {
+		return true
+	} else {
+		parentName = fmt.Sprintf("%s/%s", parentName, name)
+		paths = append(paths, parentName)
+	}
+	children := getChildren(node, parent)
+
+	for childNode := children.Oldest(); childNode != nil; childNode = childNode.Next() {
+		if walkTree(childNode.Value, parent, childNode.Key, parentName) {
+			continue
+		}
+	}
+	return true
+}
+
+func getChildren(node *jsonschema.Schema, parentSchema *jsonschema.Schema) *orderedmap.OrderedMap[string, *jsonschema.Schema] {
+	if node == nil {
+		return nil
+	}
+	if node.Ref != "" {
+		refSplit := strings.Split(node.Ref, "/")
+		fieldSchema, ok := parentSchema.Definitions[refSplit[len(refSplit)-1]]
+		if !ok {
+			panic("Couldn't find schema definition " + refSplit[len(refSplit)-1])
```

Comment:
More descriptive error message

`panic(fmt.Errorf("schema definition %q not found in reference %q", schemaName, node.Ref))`